server/task_scheduler.go

```diff
@@ -229,8 +229,21 @@ func (s *TaskScheduler) Register(taskType api.TaskType, executorGetter func() Ta
 	s.executorGetters[taskType] = executorGetter
 }
 
-// canScheduleTask checks if the task can be scheduled, i.e. change the task status from PENDING to RUNNING.
-func (s *TaskScheduler) canScheduleTask(ctx context.Context, task *api.Task) (bool, error) {
+// can transit PendingApproval -> Pending only if
+// 1. has no blocking tasks.
+// 2. task check results are all SUCCESS.
```

Comment:
Are we blocking a transition from `PENDING_APPROVAL` to `PENDING` with `WARN` task check results?

Reply:
Yes, but blocking only for the auto-approval workflow. Manually approving on the frontend doesn’t respect this.

Reply:
These functions seem to be confusing. I'll remove them and move the comments somewhere else.

Reply:
IC. So that auto-approval with warning task checks will be downgraded to manual-approval. Is that correct?

Reply:
Not quite.
Before: auto-approval policy tasks start as a Pending task.
Now: auto-approval policy tasks start as a PendingApproval task and will transit into Pending if all task checks pass.